pkg/engine/engine.go

```diff
@@ -230,9 +230,20 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 							results = detectors.CleanResults(results)
 						}
 						for _, result := range results {
-							SetResultLineNumber(chunk, &result, fragStart, mdLine)
+							resultChunk := chunk
+							if isGitSource(chunk.SourceType) {
+								copyChunk := *chunk
+								copyMetaDataClone := proto.Clone(chunk.SourceMetadata)
+								copyMetaData, ok := copyMetaDataClone.(*source_metadatapb.MetaData)
+								if ok {
+									copyChunk.SourceMetadata = copyMetaData
+								}
```

Comment:
**Nit (optional):** These lines can be combined which provides limited scope to the variables

```go
if copyMetaData, ok := copyMetaDataClone.(*source_metadatapb.MetaData); ok {
    copyChunk.SourceMetadata = copyMetaData
}
```